providers/azureblob/src/main/java/org/jclouds/azureblob/blobstore/functions/BlobPropertiesToBlobMetadata.java

```diff
@@ -58,6 +59,8 @@ public MutableBlobMetadata apply(BlobProperties from) {
             PublicAccess containerAcl = containerAcls.getUnchecked(from.getContainer());
             if (containerAcl != PublicAccess.PRIVATE)
                to.setPublicUri(from.getUrl());
+         } catch (UncheckedExecutionException ue) {
+               to.setPublicUri(from.getUrl());
```

Comment:
Would something like the following work?
```java
} catch (Exception ex) {
   InsufficientAccessRightsException iare = Throwables2.getFirstThrowableOfType(ex, InsufficientAccessRightsException.class);
   if (iare == null) throw ex; // Propagate any other uncaught exceptions
   to.setPublicUri(from.getUrl()); // handle our own exception gracefully
}
```

Reply:
Should this call `setPublicUri` when catching the exception?

Reply:
I'm not that familiar with the contract for this API and try/catch block. So... Sure, let's call `setPublicUri` wherever it makes sense. I just tried to suggest a way of capturing the _real_ exception we care about.
Once we capture the exception we want, let's make sure we set the URI when needed. I'll defer that to you, as you know the API better :)

Reply:
I agree with your comment!  But I am suggesting that if the bucket is non-public, it should not have a public URI.

Reply:
Shall we then omit setting the URI at all? What happens if this parameter is unset?

Reply:
hi @gaul! Could you please take a look at the code now? I have implemented the changes, proposed above